client/signup/utils.js

```diff
@@ -17,16 +17,18 @@ var i18nUtils = require( 'lib/i18n-utils' ),
 	formState = require( 'lib/form-state' );
 
 function getFlowName( parameters ) {
-	var currentFlowName = flows.currentFlowName;
-	if ( parameters.flowName && isFlowName( parameters.flowName ) ) {
-		return parameters.flowName;
-	}
+	const flow = ( parameters.flowName && isFlowName( parameters.flowName ) ) ? parameters.flowName : defaultFlowName;
```

Comment:
@drewblaisdell After discussing it more, we feel like we'd be weakening what we have now for little benefit. We'll merge it as-is so we can fix the Headstart test, then we can certainly keep discussing the details if we like.